.travis.yml

```diff
@@ -16,7 +16,26 @@ script:
 # following https://docs.travis-ci.com/user/environment-variables/#convenience-variables
 # the pull request isn't set to true if its a PR; its just set to false if its not.
 - if [ "$TRAVIS_PULL_REQUEST" != "false" ] && [ "$TRAVIS_BRANCH" == "master" ]; then bash ./scripts/validate-version-bump.sh; fi
-
+after_success:
+- |
+  # PSEUDO CODE
+  if branch == master
+      prev_tag = git tag -l | HEAD -1
+      msg = $TRAVIS_COMMIT_MESSAGE
+      if msg.startswith('[patch]')
+          bump = 0.0.1
+      elif msg.startswith('[minor]')
+          bump = 0.1.0
+      elif msg.startswith('[major]')
+          bump = 1.0.0
+      elif msg.startswith('[pre]')
+          bump = None
+      else
+          exit 1 "Invalid semver bump"
+      if bump
+          tag = prev_tag + bump
+          git tag $tag
+          git push --tags
```

Comment:
Note that this is effectively going back to continuous deployment of every merge to master.  Currently, our release process involves manually triggering verification jobs before cutting releases; If we make this change, we probably need to gate PRs to master on those tests against the PR code, which could significantly add to jenkins load.